BeerOverflowWindowsApp/BeerPanel.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Drawing;
+using System.Data;
+using System.Text;
+using System.Windows.Forms;
+using System.Threading;
+
+namespace BeerOverflowWindowsApp
+{
+    public class BeerPanel : System.Windows.Forms.Panel
+    {
+        Font textFont = new Font("Times New Roman", 24);
+        private Point top;
+        private Point bottom;
+        private Rectangle rectangle;
+        private int height;
+        private static string rating;
+        public static string Rating
+        {
+            get { return rating; }
+        }
+        private const int markerHeight = 30;
+        private const int markerWidth = 50;
+
+        public BeerPanel(int topX, int topY, int height)
+        {
+            this.DoubleBuffered = true;
+            InitializeComponent();
+            top = new Point(topX, topY);
+            this.height = height;
+            bottom = new Point(top.X, top.Y + height);
+            rectangle = new Rectangle(bottom.X - markerWidth / 2, bottom.Y - markerHeight / 2, markerWidth, markerHeight);
+        }
+
+        private void BeerPanel_Paint(object sender, PaintEventArgs e)
+        {
+            Paint_Beer_Meter(e.Graphics);
+        }
+        private void InitializeComponent()
+        {
+            this.SuspendLayout();
+            // 
+            // BeerPanel
+            // 
+            this.BackColor = System.Drawing.SystemColors.Window;
+            this.Paint += new System.Windows.Forms.PaintEventHandler(this.BeerPanel_Paint);
+            this.MouseDown += new System.Windows.Forms.MouseEventHandler(this.BeerPanel_MouseDown);
+            this.MouseMove += new System.Windows.Forms.MouseEventHandler(this.BeerPanel_MouseMove);
+            this.MouseUp += new System.Windows.Forms.MouseEventHandler(this.BeerPanel_MouseUp);
+            this.ResumeLayout(false);
+        }
+
+        private String text = "0 %";
+        private void Paint_Beer_Meter(Graphics g)
+        {
+            g.DrawLine(System.Drawing.Pens.Black, top, bottom);
+            g.DrawEllipse(System.Drawing.Pens.Black, rectangle);
+            g.DrawString(text, new Font("Arial", 16), new SolidBrush(Color.Black), top.X - 30 , top.Y - 45);
+        }
+
+        int difference = 0;
+        private Boolean mousePressed = false;
+        private void BeerPanel_MouseDown(object sender, MouseEventArgs e)
+        {
+            if (rectangle.Contains(e.X, e.Y))
+            {
+                difference = e.Y - rectangle.Y;
+                mousePressed = true;
+            }
+        }
+
+        private void BeerPanel_MouseMove(object sender, MouseEventArgs e)
```

Comment:
Can you explain the math you're doing in this function?

Reply:
First of all, all calculations are done only with Y coordinates. The "rectangle" variable actually represents the position of the marker(going to rename that). "rectangle.Y + rectangle.Height / 2" is used to determine the Y coordinate of the middle of the marker and those two if's are just for not letting the marker to go above "top" point or below "bottom" point. 
"rating = (bottom.Y - rectangle.Y - rectangle.Height / 2) / (height / 100)" is calculating the percentage value of how much of the glass is filled (beer height / glass height * 100 just in a little bit different form)